api/types.go

```diff
@@ -157,6 +157,23 @@ func (tr Timerange) IsValid() bool {
 		tr.Start <= tr.End
 }
 
+// Round() will fix some invalid timeranges by rounding their starts and ends.
+func (tr Timerange) Round() Timerange {
+	if tr.Resolution == 0 {
+		return tr
+	}
+	tr.Start = (tr.Start + tr.Resolution/2) / tr.Resolution * tr.Resolution // This performs a round
+	tr.End = (tr.End + tr.Resolution/2) / tr.Resolution * tr.Resolution     // This performs a round
+	return tr
+}
+
+// Later() returns a timerange which is forward in time by the amount given
+func (tr Timerange) Later(time int64) Timerange {
```

Comment:
Yes, that's important to maintain. In terms of a proof of correctness or similar, if you consider:

`start` and `end` are the two variables, both multiples of `resolution` (short `res`) since the input must be well-formed, so let `start = res*s` and `end = res*e`, so our rounding ends up with `(res*s + shift + res/2) / res * res` for the new start, and `(res*e + shift + res/2)/res*res` for our new end (all divisions are integer division).

`(res*s + shift + res/2)/res*res = (res*s)/res*res + (shift + res/2)/res*res` since `(res*s)` is an exact multiple of `res`, so we can distribute the division, and then `= (s*res)/res*res + (shift + res/2)/res*res = s*res + (shift + res/2)/res`.

Similarly, the new end will be `e*res + (shift + res/2)/res`. So their differences will still be `e*res - s*res` which is exactly `end - start`.

Having said that, test cases would be good and I can add them to api.types to verify this behavior.